bokehjs/src/coffee/models/grids/grid.ts

```diff
@@ -194,6 +195,11 @@ export class Grid extends GuideRenderer {
       //  end = null
     } else {
       [start, end] = range_bounds
+      for (const axis of this.plot.select(Axis)) {
```

Comment:
For now you have to assert the expected type with `for (const axis of this.plot.select(Axis) as Axis[])`.